server/planning/feeding_services/event_email_service.py

```diff
@@ -117,7 +121,7 @@ def _update(self, provider, update):
                                                 content = io.BytesIO(attachment)
                                                 res = process_file_from_stream(content, part.get_content_type())
                                                 file_name, content_type, metadata = res
-                                                if isinstance(parser, NTBEventXMLFeedParser):
+                                                if NTBEventXMLFeedParser and isinstance(parser, NTBEventXMLFeedParser):
```

Comment:
Instead of attempting to import the `NTBEventXMLFeedParser` and testing `isinstance`, I think it might be better to use something similar to `can_parse` of the registered feed parser itself. This way these 3 usages for both `NTBEventXMLFeedParser` and `IcsTwoFeedParser` can be removed and move the logic into the parser themselves.

Reply:
The logic we have right now in `event_file_service.py`, `event_http_service.py` and `event_email_service.py` checks a class anyway. I don't see a way how a method in a parser can help here. We can avoid attempting to import `NTBEventXMLFeedParser` and import 'IcsTwoFeedParser' by checking a class name `if parser.__class__.__name__ == 'NTBEventXMLFeedParser'` which is IMO a less reliable.
Can you provide me an example of how you see a method similar to `can_parse ` ?

Reply:
Hi @MarkLark86 , any update on it :arrow_up: ?

Reply:
You can use python's `getattr` to determine if the parser has a specific method defined, otherwise use the default. The purpose is to remove all specific NTB/ICS code from these services.

You could use something like the following:
```python
if getattr(parser, 'parse_email'):
    parser.parse_email(....)
else:
    logger.warn('Ingesting event with unknown parser'
    new_items.append(parser.parse(data, provider)
```

And then implement something similar to `parse_file` and `parse_http`.

This will make it easier to maintain the code by separating out the implementation of the parser from the service entirely.

Reply:
@MarkLark86 check it, please. 
Сorresponding сommit for NTB https://github.com/superdesk/superdesk-ntb/pull/344/commits/9d213172b0b21d62a11e9a122917f2806665b3ce

Reply:
Good to go